### ISSUE TITLE
[Bug] Some modules fail with ZOAU config errors

### ISSUE THREAD

**Issue by petelouis** [ansible-collections/ibm_zos_core#266]
##### SUMMARY

Both zos_lineinfile and zos_operator_action_query are failing with the following error:
fatal: [tvt5033.svl.ibm.com]: FAILED! => {"changed": false, "msg": "An unexpected error occurred: ImportError('ZOAU is not properly configured for Ansible. Unable to import zoautil_py. Ensure environment variables are properly configured in Ansible for use with ZOAU.')"}

As far as I can tell, I have ZOAU configured as described in the doc, and I have other core modules working, such as zos_fetch, zos_operator, zos_job_submit, etc., which seems to imply that it's configured correctly.  My environment variables look like the following, am I missing something?
  environment:
    PATH: "/usr/lpp/IBM/cyp/v3r8/pyz/bin:/bin:/usr/lpp/java/J8.0_64/bin:/usr/lpp/IBM/zoautil/bin" 
    LIBPATH: "/usr/lib/java_runtime:/usr/lpp/IBM/cyp/v3r8/pyz/lib:/usr/lpp/IBM/zoautil/lib" 
    PYTHONPATH: "/usr/lpp/IBM/cyp/v3r8/pyz/lib:/usr/lpp/IBM/zoautil/lib" 
    CLASSPATH: "/usr/lpp/IBM/zoautil/lib/*"
    JAVA_HOME: "/usr/lpp/java/J8.0_64"
    _BPXK_AUTOCVT: "ON"
    LANG: "C"
    _CEE_RUNOPTS: "FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"
    _TAG_REDIR_ERR: "txt"                                
    _TAG_REDIR_IN: "txt"
    _TAG_REDIR_OUT: "txt"

examples of my failing tasks are:

- name: review sms validate listing
  zos_lineinfile:
    src: SMS.VALIDATE.LISTING
    line: 'validation verified'

- name: check_wtor
  zos_operator_action_query:
    message_id: DFS*

##### ANSIBLE VERSION

```
Ansible version output:
ansible 2.9.15
  config file = /home/plouis/ansible/ansible.cfg
  configured module search path = ['/home/plouis/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.8 (default, Aug 18 2020, 08:33:21) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]

```

##### SPECIFY ANSIBLE COLLECTION VERSION

  "version": "1.3.0-beta.2",

##### SPECIFY THE Z OPEN AUTOMATION UTILITIES VERSION

2020/07/07 14:54:31 CUT V1.0.3  

##### ENVIRONMENT

Ansible running on RHEL 8.3, z/OS is 2.4 at RSU2009

##### STEPS TO REPRODUCE

```yaml
Reproduction playbook/yaml
```

##### ACTUAL RESULTS

```
Playbook output:
```

#### Ansible.cfg

```
'ansible-config dump --only-changed' output:
```

#### Inventory

```
Inventory content:
```

#### Vars

```
'group_vars' or 'host_vars' content:
```

**Comment by ddimatos** [comment]
Hi @petelouis - thanks for opening this up, we were a bit distracted with releases and planning and thus did not have time to review this till today. 

So, not all modules rely on ZOAU hence some of your modules such as `zos_copy` and `zos_fetch` would have executed correctly.

I see a couple of issues, the first being the collection "version": "1.3.0-beta.2" requires ZOAU 1.1.0 or ZOAU 1.1.1; we just promoted the `1.3.0-beta.2` to `ibm_zos_core v1.3.0` (same requirements) which is now our latest stable GA, [see the requirements](https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/release_notes.html#reference). 

After a promotion, I sunset the beta docs but they can still be found in the gitHub versions [here](https://github.com/ansible-collections/ibm_zos_core/blob/v1.3.0-beta.2/docs/source/release_notes.rst#reference). You either need to reduce your Ansbile version to something supported by ZOAU 1.0.3 or update your ZOAU version to either 1.1.0 or 1.1.1 of which you can obtain an unsupported version of ZOAU from [here](https://ibm.github.io/mainframe-downloads/downloads.html#devops) (search on z Open Automation Utilities to find the download).

Do note there is a PIP install for ZOAU 1.1.0 python libraries, you may prefer the latest ZOAU 1.1.1 that give you a choice of how you want to access the Python libraries (PIP or environment variable PYTHONPATH: "{{ ZOAU }}/lib") . 

The second thing I noticed is that it does look like your vars are not configured properly, see [here](https://github.com/IBM/z_ansible_collections_samples/blob/master/docs/share/configuration_guide.md) how to configure. I immediately noticed you are missing ZOAU_HOME. You don't need JAVA as part of these but it will not cause any issues.

You can simply set Python and ZOAU and use the snippet below:
```
PYZ: "/usr/lpp/IBM/cyp/v3r8/pyz"
ZOAU: "/usr/lpp/IBM/zoautil"

environment_vars:
  _BPXK_AUTOCVT: "ON"
  ZOAU_HOME: "{{ ZOAU }}"
  PYTHONPATH: "{{ ZOAU }}/lib"
  LIBPATH: "{{ ZOAU }}/lib:{{ PYZ }}/lib:/lib:/usr/lib:."
  PATH: "{{ ZOAU }}/bin:{{ PYZ }}/bin:/bin:/var/bin:/usr/sbin"
  _CEE_RUNOPTS: "FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"
  _TAG_REDIR_ERR: "txt"
  _TAG_REDIR_IN: "txt"
  _TAG_REDIR_OUT: "txt"
  LANG: "C"
```

TL;DR 
Update your ZOAU to 1.1.1 and fix the vars.

**Comment by ddimatos** [comment]
One more note, I am planning to release a minor update this week `ibm_zos_core v1.3.1`, it will have a change to support Red Hats documentation generation only thus our ssh connection plugin and zos_ping module will be affected so its up to you if you want to wait for that. Nothing else changes, no dependencies change.

**Comment by petelouis** [comment]
After switching to Z0AU 1.1.1, I initially got the following error:
BGYSC0236E Unable to call authorized program IKJEFT01 from mvscmd. Use mvscmdauth.
I fixed this by setting the APF authorized bit on in the mvscmdauth extended attributes.  Once I did this, everything now appears to be working correctly.  Maybe the ZOAU folks need to double check the extended attribute bits in the 1.1.1 beta package?

**Comment by ddimatos** [comment]
@petelouis - when you expanded ZOAU 1.1.1 did you have the authority to do so such that archive retained its extended attribute bits? I can pull the version for download and try again to be sure. Also, note that its two files that need APF, see my comment [here](https://github.com/ansible-collections/ibm_zos_core/issues/258#issuecomment-817025723).

**Comment by petelouis** [comment]
Well, I have all the authority I need, but the download didn't come with any installation instructions.  Maybe I missed an option on the pax command?

pax -rvf /u/louis/zoau/zoau-1.1.1.pax

**Comment by ddimatos** [comment]
Command I use is right out of the docs `pax -p e -rf zoau-<version>.pax` , docs are [here](https://www.ibm.com/docs/en/zoau/1.1.1?topic=installing-configuring-zoa-utilities), I still want to double check it but never had an issue with the above command preserving the attributes when I expanded as UID 0. 

The `e` is [documented as such](https://www.ibm.com/docs/en/zos/2.2.0?topic=descriptions-pax-interchange-portable-archives):

```
Preserves the user ID, group ID, file mode, access time, modification time, extended attributes, and ACL entries. Prior to z/OS
1.8, audit flags and file format (line end) attributes were not restored because they are not available in any archive format. The
extended attributes are the apsl flags that are set by the extattr command. A pax format archive can be used to store the 
audit flags and file format, and -p e will restore them when available.
```

The `p` as:
```
Preserves the file mode: access permissions (without modification by umask), set-user-ID bit, set-group-ID bit, and sticky bit.

**pax** restores access permissions by default. If _UNIX03=YES extracted files will have permissions of 0666 (modified by umask) unless -p p or -p e are used. 
```

Looks like you are missing the options `-p e` that preserves the extended attributes needed for authorization. 

I will consult with my team if we should include a README that points back to the doc.

**Comment by petelouis** [comment]
You are correct, I failed to check the zoau doc.  When I use the pax command from the doc, all the bits are set correctly and everything works fine.  Feel free to close this one whenever you're ready.

**Comment by ddimatos** [comment]
Glad it worked out, thanks again for engaging us.
